### PR TITLE
Make sure that object id can be a string

### PR DESCRIPTION
**Pull request by nickvergessen**
@DeepDiver1975 @MorrisJobke @rullzer preparation for new share and file ids

### CODE REVIEW

**Comment by nickvergessen** [comment]
App patch https://github.com/owncloud/notifications/pull/49

File: lib/private/notification/notification.php [hunk 1]
```diff
@@ -175,10 +176,10 @@ public function setObject($type, $id) {
 		}
 		$this->objectType = $type;
 
-		if (!is_int($id)) {
+		if (!is_int($id) && (!is_string($id) || $id === '' || isset($id[64]))) {
```
Comment (rullzer) [inline lib/private/notification/notification.php]: Out of curiosity why 64 chars? Why

**Comment by rullzer** [comment]
:+1:

File: lib/private/notification/notification.php (hunk 1 above)
Comment (rullzer) [inline lib/private/notification/notification.php]: A nevermind... the db just holds 64 chars :P

Comment (nickvergessen) [inline lib/private/notification/notification.php]: So we have a maximum that we can use for the database table?

**Comment by PVince81** [comment]
:+1:

**Pull request merged**